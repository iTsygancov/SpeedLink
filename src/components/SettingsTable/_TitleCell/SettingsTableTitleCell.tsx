import { Input } from "@/components/ui/input";
import { TableCell } from "@/components/ui/table";
import { Shortcut } from "@/types";

export type SettingsTableTitleCellProps = {
  item: Shortcut;
  itemIndex: number;
  handleChangeShortcut: (
    event: React.ChangeEvent<HTMLInputElement>,
    itemIndex: number
  ) => void;
};

const SettingsTableTitleCell = ({
  item,
  itemIndex,
  handleChangeShortcut
}: SettingsTableTitleCellProps) => {
  return (
    <TableCell className='p-0 pl-4 text-left'>
      {item.canEdit ? (
        <Input
          value={item.title}
          name='title'
          onChange={(event) => handleChangeShortcut(event, itemIndex)}
        />
      ) : (
        item.title || "-"
      )}
    </TableCell>
  );
};

export default SettingsTableTitleCell;
