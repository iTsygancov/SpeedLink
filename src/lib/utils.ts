import { type ClassValue, clsx } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export function generateAlphanumericKeysArray() {
  const keysArray = [];

  // Generate lowercase alphabetic keys
  for (let i = 65; i <= 90; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  // Generate numeric keys
  for (let i = 48; i <= 57; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  return keysArray;
}
