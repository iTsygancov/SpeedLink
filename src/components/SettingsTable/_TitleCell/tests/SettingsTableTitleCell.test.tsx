import SettingsTableTitleCell, {
  SettingsTableTitleCellProps
} from "../SettingsTableTitleCell";
import { initialProps } from "./SettingsTableTitleCell.data";
import { expectToMatchSnapshot, renderTree } from "@/test/utils";
import { describe, it } from "vitest";

const testComponentWithProps = (props: SettingsTableTitleCellProps) => {
  expectToMatchSnapshot(renderTree(<SettingsTableTitleCell {...props} />));
};

describe("SettingsTableTitleCell", () => {
  it("should render with empty title", () => {
    testComponentWithProps(initialProps);
  });

  it("should render with filled title", () => {
    testComponentWithProps({
      ...initialProps,
      item: {
        ...initialProps.item,
        title: "Test title"
      }
    });
  });

  it("should render with empty input field", () => {
    testComponentWithProps({
      ...initialProps,
      item: {
        ...initialProps.item,
        canEdit: true
      }
    });
  });

  it("should render with filled input field", () => {
    testComponentWithProps({
      ...initialProps,
      item: {
        ...initialProps.item,
        canEdit: true,
        title: "Test title"
      }
    });
  });
});
