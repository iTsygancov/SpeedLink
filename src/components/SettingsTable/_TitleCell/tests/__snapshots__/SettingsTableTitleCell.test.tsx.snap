// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsTableTitleCell > should render with empty input field 1`] = `
<td
  className="align-middle [&:has([role=checkbox])]:pr-0 p-0 pl-4 text-left"
>
  <input
    className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
    name="title"
    onChange={[Function]}
    value=""
  />
</td>
`;

exports[`SettingsTableTitleCell > should render with empty title 1`] = `
<td
  className="align-middle [&:has([role=checkbox])]:pr-0 p-0 pl-4 text-left"
>
  -
</td>
`;

exports[`SettingsTableTitleCell > should render with filled input field 1`] = `
<td
  className="align-middle [&:has([role=checkbox])]:pr-0 p-0 pl-4 text-left"
>
  <input
    className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
    name="title"
    onChange={[Function]}
    value="Test title"
  />
</td>
`;

exports[`SettingsTableTitleCell > should render with filled title 1`] = `
<td
  className="align-middle [&:has([role=checkbox])]:pr-0 p-0 pl-4 text-left"
>
  Test title
</td>
`;
